src/coreComponents/constitutive/fluid/multifluid/CO2Brine/CO2BrineFluid.cpp

```diff
@@ -137,6 +138,50 @@ integer CO2BrineFluid< PHASE1, PHASE2, FLASH >::getWaterPhaseIndex() const
   return PVTFunctionHelpers::findName( m_phaseNames, expectedWaterPhaseNames, viewKeyStruct::phaseNamesString() );
 }
 
+template< typename PHASE1, typename PHASE2, typename FLASH >
+void CO2BrineFluid< PHASE1, PHASE2, FLASH >::checkTablesParameters( real64 const pressure,
+                                                                    real64 const temperature ) const
+{
+  if( !m_checkPVTTablesRanges )
+  {
+    return;
+  }
+
+  real64 const temperatureInCelsius = constants::convertKToC( temperature );
+  try
+  {
+    m_phase1->density.checkTablesParameters( pressure, temperatureInCelsius );
+    m_phase1->viscosity.checkTablesParameters( pressure, temperatureInCelsius );
+    m_phase1->enthalpy.checkTablesParameters( pressure, temperatureInCelsius );
+  } catch( SimulationError const & ex )
+  {
+    // TODO: replace getName() by getDataContext() from incoming PR 2404
+    string const errorMsg = GEOS_FMT( "{}: Table input error for phase no. 1.\n", getName() );
+    throw SimulationError( ex, errorMsg );
+  }
+  try
+  {
+    m_phase2->density.checkTablesParameters( pressure, temperatureInCelsius );
+    m_phase2->viscosity.checkTablesParameters( pressure, temperatureInCelsius );
+    m_phase2->enthalpy.checkTablesParameters( pressure, temperatureInCelsius );
+  } catch( SimulationError const & ex )
+  {
+    // TODO: replace getName() by getDataContext() from incoming PR 2404
+    string const errorMsg = GEOS_FMT( "{}: Table input error for phase no. 2.\n", getName() );
+    throw SimulationError( ex, errorMsg );
+  }
+  try
+  {
+    m_flash->checkTablesParameters( pressure, temperatureInCelsius );
+  } catch( SimulationError const & ex )
+  {
+    // TODO: replace getName() by getDataContext() from incoming PR 2404
+    string const errorMsg = GEOS_FMT( "{}: Table input error for flash phase.\n", getName() );
+    throw SimulationError( ex, errorMsg );
+  }
```

Comment:
Would any possible error message due to temperature range violation come out in degC which might be confusing since user input is in K?

Reply:
@dkachuma Indeed, the messages were in C°, I changed them to K°. I proceeded by adding:
- units metadata to the `TableFunction` class,
- units functions to properly format values for the user.